.github/workflows/release.yml

```diff
@@ -1,28 +1,126 @@
+
+  publish-tag:
+    name: Publish tag
+    needs: check-permission
+    if: contains(needs.check-permission.outputs.hasWritePermission, 'true')
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout prebid cache
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Create & publish tag
+        id: release
+        run: |
+          currentTag=$(git describe --abbrev=0 --tags)
+          echo "Current release tag ${currentTag}"
+
+          echo ${currentTag} | grep -q "^v\?[0-9]\+\.[0-9]\+\.[0-9]\+$"
+          if [ $? -ne 0 ]; then
+            echo "Current tag format won't let us compute the new tag name. Required format v[0-9]\+\.[0-9]\+\.[0-9]\+"
+            exit 1
+          fi
+
+          if [[ "${currentTag:0:1}" != "v" ]]; then
+            currentTag="v${currentTag}"
+          fi
+
+          nextTag=''
+          releaseType=${{ inputs.releaseType }}
+          if [ $releaseType == "minor" ]; then
+            # increment minor version and reset patch version
+            nextTag=$(echo "${currentTag}" | awk -F. '{OFS="."; $2+=1; $3=0; print $0}')
+          else
+            # increment patch version
+            nextTag=$(echo "${currentTag}" | awk -F. '{OFS="."; $3+=1; print $0}')
+          fi
+
+          if [ ${{ inputs.debug }} == 'true' ]; then
+            echo "running workflow in debug mode, next ${releaseType} tag: ${nextTag}"
+          else
+            git tag $nextTag
+            git push origin $nextTag
+            echo "tag=${nextTag}" >> $GITHUB_OUTPUT
+          fi
+    outputs:
+      releaseTag: ${{ steps.release.outputs.tag }}
+
+  publish-docker-image:
+    name: Publish docker image
+    needs: publish-tag
+    if: contains(inputs.debug, 'false')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout prebid cache
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Build image
         run: |
-          echo "tag=${GITHUB_REF/refs\/tags\/}" >> $GITHUB_OUTPUT
-          echo "version=${GITHUB_REF/refs\/tags\/v}" >> $GITHUB_OUTPUT
+          docker build --build-arg DEPLOY_VERSION=${{ needs.publish-tag.outputs.releaseTag }} -t docker.io/prebid/prebid-server:${{ needs.publish-tag.outputs.releaseTag }} .
+      - name: Login to docker Hub
+        if: contains(inputs.debug, 'false')
+        uses: docker/login-action@v2.1.0
+        with:
+          username: ${{ secrets.DOCKERHUB_USER }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+      - name: Publish to docker Hub
+        run: |
+          docker push docker.io/prebid/prebid-server:${{ needs.publish-tag.outputs.releaseTag }}
 
-      - name: Create & Publish Release
-        uses: release-drafter/release-drafter@v5.12.1
+  publish-release:
+    name: Publish release
+    needs: [publish-tag, publish-docker-image] 
+    if: contains(inputs.debug, 'false')
```

Comment:
For my understanding, How does if works here?

Reply:
```
<img width="256" height="128" alt="Screenshot 2023-05-11 at 3 15 51 PM" src="https://github.com/prebid/prebid-server/assets/24757781/8f14b0f0-81ce-44f8-8ebd-04e86aac27a7">

We are providing user with checkbox to run script in debug mode.

```
  workflow_dispatch:
    inputs:
      releaseType:
        type: choice
        ....
      debug:
        type: boolean
        default: true
```

Above lines are declared at the start of script. State of checkbox i.e checked or unchecked is tracked in `inputs.debug` var.

`if: contains(inputs.debug, 'false')` this line checks if checkbox is being checked. If not then publish release notes